Sources/client/Socket.swift

```diff
@@ -147,51 +150,37 @@ public class Socket {
   //----------------------------------------------------------------------
   // MARK: - Initialization
   //----------------------------------------------------------------------
+  public convenience init(_ endPoint: String) {
+    self.init(endPoint: endPoint,
+              transport: { url in return WebSocket(url: url) },
+              params: { nil })
+  }
+
   public convenience init(_ endPoint: String,
-                          params: [String: Any]? = nil) {
+                          params: Payload?) {
```

Comment:
I would prefer to see this init signature stay like `params: Payload? = nil` and then change `paramsGetter: PayloadGetterFunction?` to not have a default `nil`. Then we can remove the added `convenience init` and not introduce a change to the client's current API

Reply:
What do you do if you provide both?

Socket("https://somewhere", params: ["hello":"world"], { ["hello":"another world", "and_more": "stuff"] })

This isn't clear to me, and the API is not very clear if this is the way to go. If Socket(_: String, params: [:]) instead causes a deprecation warning and we introduce Socket(_: String, paramsClosure: [:]) that's a lot clearer.

That's the reason I did what I did.

Reply:
Is this a valid case? Likely a client that used a `PayloadClosure` will provide `PayloadClosure` that returns the correct initial payload and client that provides just `Payload` will not need to use the dynamic closure.

 I wouldn't say deprecating `params: Payload?` init is necessary